rotkehlchen/tests/api/test_assets.py

```diff
@@ -262,3 +262,150 @@ def test_ignored_assets_endpoint_errors(rotkehlchen_api_server_with_exchanges, m
         )
         # Check that assets did not get modified
         assert set(rotki.data.db.get_ignored_assets(cursor)) >= expected_tokens
+
+
+def test_get_all_assets(rotkehlchen_api_server):
+    """Test that check fetching all assets returns a paginated result."""
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'allassetsresource',
+        ),
+        json={
+            'limit': 20,
+            'offset': 0,
+            'asset_type': 'fiat',
+            'order_by_attributes': ['name'],
+            'ascending': [True],
+        },
+    )
+    result = assert_proper_response_with_result(response)
+    assert len(result['entries']) == 20
+    assert 'entries_found' in result
+    assert 'entries_total' in result
+    assert 'entries_limit' in result
+
+    # check that providing multiple order_by_attributes fails
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'allassetsresource',
+        ),
+        json={
+            'limit': 20,
+            'offset': 0,
+            'asset_type': 'fiat',
+            'order_by_attributes': ['name', 'symbol'],
+            'ascending': [True, False],
+        },
+    )
+    assert_error_response(response, contained_in_msg='Multiple fields ordering is not allowed.')
+
+
+def test_get_assets_mappings(rotkehlchen_api_server):
+    """Test that providing a list of asset identifiers, the appropriate assets mappings are returned."""  # noqa: E501
+    queried_assets = ('BTC', 'TRY', 'EUR')
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'assetsmappingresource',
+        ),
+        json={'identifiers': queried_assets},
+    )
+    result = assert_proper_response_with_result(response)
+    assert len(result) == 3
+    for entry in result:
+        assert entry['identifier'] in queried_assets
+
+    # check that providing multiple order_by_attributes fails
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'assetsmappingresource',
+        ),
+        json={'identifiers': ['BTC', 'TRY', 'invalid']},
+    )
+    assert_error_response(response, contained_in_msg='One or more identifiers do not exist')
+
+
+def test_search_assets(rotkehlchen_api_server):
+    """Test that searching for assets using a keyword works."""
+    response = requests.post(
```

Comment:
would need to test different limits, test using search by name, search by symbol, different ordering